torch/csrc/jit/codegen/cuda/utils.h

```diff
@@ -75,6 +78,158 @@ class PolymorphicBase {
   }
 };
 
+//! Container class DisjointSet models equivalence relationships
+//!
+//! Each instance of this class keeps a set of equivalent classes
+//! DisjointSet::join(a,b) makes the full class of a and b equivalent
+//! DisjointSet::areEqual(a,b) checks if a and b belong same class
+//!
+//! \note The template type T is assumed to be hashable
+template <typename T, typename Hash=std::hash<T>>
+class DisjointSet {
+ public:
+  DisjointSet() = default;
+
+  //! Joins the equivalent class that a and b belong to
+  //! areEqual(a',b') will be true for each a'=a and b'=b
+  //!
+  //! \param a An element from a equivalent class
+  //!          will create a new equivalent class if a does
+  //!          not belong to any
+  //! \param b An element from another equivalent class
+  //!          will create a new equivalent class if b does
+  //!          not belong to any
+  void join(T a, T b) {
```

Comment:
Is this called anywhere anymore?

Reply:
Yes. It's used when two objects of type `T` should be grouped into the same set. In RootDomainMap, it is used when two `IterDomain`s are mappable each other. `RootDomainMap::proveId` calls it.